qiskit/transpiler/passes/routing/bip_mapping.py

```diff
@@ -105,18 +105,20 @@ def __init__(
                 pip_install="pip install 'qiskit-terra[bip-mapper]'",
             )
         super().__init__()
-        self.coupling_map = copy.deepcopy(coupling_map)  # save a copy to modify
+        self.global_coupling_map = coupling_map
+        self.coupling_map = copy.deepcopy(coupling_map)  # copy to modify
+        self.global_qubit = qubit_subset
         if self.coupling_map is not None:
+            if self.global_qubit is None:
+                self.global_qubit = list(range(self.coupling_map.size()))
+            else:
+                self.coupling_map = self.coupling_map.reduce(self.global_qubit)
```

Comment:
what is the difference between `.reduce()` and `.subgraph()` methods here?

Reply:
#6738 fixes a bug on .subgraph() btw

Reply:
`subgraph()` does not retain the order of passed qubits while `reduce()` does. For example (copied from https://github.com/Qiskit/qiskit-terra/issues/6736#issuecomment-881557566),
```
coupling = CouplingMap.from_line(6, bidirectional=False)
list(coupling.subgraph([4, 2, 3, 5]).get_edges())
>>> [(0, 1), (1, 2), (2, 3)]
list(coupling.reduce([4, 2, 3, 5]).get_edges())
>>> [(1, 2), (2, 0), (0, 3)]
```
We need the map from reduced qubits (in coupling_map) to global qubits (in global_coupling_map) to recover the global qubits from reduced qubits. Using `reduce()`, the map can be naturally represented as the list `qubit_subset` (and that's why the type of `qubit_subset` is `list` (not `set`) here).